pkg/cache/nodes.go

```diff
@@ -213,21 +212,22 @@ func (nc *schedulerNodes) updateNode(oldNode, newNode *v1.Node) {
 		return
 	}
 
-	if schedulerNode, ok := nc.nodesMap[newNode.Name]; ok {
-		schedulerNode.ready = common.HasReadyCondition(newNode)
+	schedulerNode, ok := nc.nodesMap[newNode.Name]
+	if ok {
+		schedulerNode.ready = hasReadyCondition(newNode)
```

Comment:
The logic here changed slightly. We only send out the update to the proxy if the node is found in the `nodesMap`. I don't think it has substantial effects, just wondering if this is better?

cc @wilfred-s

Reply:
Yes, it should not have any effects. I don't see a situation wherein entry doesn't exist in `nodesMap `during "update"

Reply:
I do think we need to carefully look at what we do here (referencing new line numbers):
* line 195: we get the node out of the nodesMap store in cachedNode
* line 215: we retrieve the same node again, we already know it CANNOT be nil and ok is `true, we should re-use cachedNode
* line 217: should always be executed, even if resources do not change, should this use a locked setter method?